### PR TITLE
Organize index of guides into 3 categories

### PR DESCRIPTION
**Pull request by pastelmind**
Split the toctree of guides into three broad categories:

- Installing Packages
    - Includes guides for installing and managing dependencies.
- Building and Publishing Projects
    - Includes guides for project configuration, distribution
- Miscellaneous
    - Guides that do not clearly fit into any of the above

This commit primarily affects the Guides index page; it does not create nested lists for each category in the sidebar.

### CODE REVIEW

**Review by webknjaz — APPROVED** [review]

File: source/guides/index.rst [hunk 1]
```diff
@@ -23,10 +28,16 @@ introduction to packaging, see :doc:`/tutorials/index`.
...
    publishing-package-distribution-releases-using-github-actions-ci-cd-workflows
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Miscellaneous:
+
+   tool-recommendations
+   analyzing-pypi-package-downloads
+   index-mirrors-and-caches
+   hosting-your-own-index
```
Comment (di) [inline source/guides/index.rst:43]: I think these two should be under "installing".

Comment (pastelmind) [reply source/guides/index.rst:43]: Good point. If I update the PR, do I have to wait another 9 months to get another review?

Comment (pradyunsg) [reply source/guides/index.rst:43]: No.

**Comment by pastelmind** [comment]
@di Moved the two items. I'm sorry about making a snarky comment earlier.

**Comment by pradyunsg** [comment]
Comparing the sidebar between https://packaging.python.org/ and https://python-packaging-user-guide--708.org.readthedocs.build/, I'd prefer to keep the sidebar unchanged.

Is there some change that I'm missing?

**Comment by di** [comment]
@pradyunsg Looks like that was due to the branch being out of date.

**Comment by di** [comment]
Thanks @pastelmind!

**Pull request merged**